src/relayclient/GSNConfigurator.ts

```diff
@@ -33,6 +33,22 @@ const defaultGsnConfig: GSNConfig = {
   clientId: '1'
 }
 
+export async function resolveGSNDeploymentFromPaymaster (provider: provider, partialConfig: Partial<GSNConfig>): Promise<Partial<GSNConfig>> {
```

Comment:
something like this, except that you want to call it automatically when there is no RelayHub configured